### DISCUSSION TITLE
Validate Tiptap schema on backend server side with Node?

### DISCUSSION THREAD

**Discussion by nabramow** [ueberdosis/tiptap#4454] [General Questions]
I'm trying to find a way to validate tiptap content against the schema on the backend using Node. For example, if someone were to send an api call to us mocking tiptap data and adding something malicious. In this case tiptap's validation wouldn't help.

I've been searching google and the tiptap and prosemirror docs and repos for hours and can't find the best way to do this. Closest example I found is this reference related to ProseMirror: https://github.com/ueberdosis/tiptap/discussions/241

Has anyone done this and have any tips? Is there something I'm missing and that's why nothing comes up when I search? Thank you!

**Comment by ProTip** [comment]
Here is how you can get access to the schema: https://tiptap.dev/api/schema#option-2-without-an-editor .

This will just be a ProseMirror schema. So if you try to create an [EditorState](https://prosemirror.net/docs/ref/) (It's right there at the top) or even just [construct a Node with fromJSON](https://prosemirror.net/docs/ref/#model.Node) with it in Node.js it will fail if the content violates the schema.

**Reply by nabramow** [reply]
Thank you! We ended up just deciding to sanitize the HTML rather than check against the schema as it was going to involve importing way more packages. In case that info is useful to anyone else reading this later!
